lib/crypto/index.js

```diff
@@ -298,30 +303,79 @@ Crypto.prototype.downloadKeys = function(userIds, forceDownload) {
         return q(stored);
     }
 
-    return this._baseApis.downloadKeysForUsers(
+    var r = this._doKeyDownloadForUsers(downloadUsers);
+    var promises = [];
+    downloadUsers.map(function(u) {
```

Comment:
`Array.forEach` doesn't exist pre-ES5, and I don't think we have a polyfill for it. Is there a reason to avoid `map` ?